bpf/lib/ipv4.h

```diff
@@ -86,14 +87,18 @@ static __always_inline bool ipv4_is_not_first_fragment(const struct iphdr *ip4)
 }
 
 static __always_inline int
-ipv4_frag_get_l4ports(const struct ipv4_frag_id *frag_id,
+ipv4_frag_get_l4ports(struct __ctx_buff *ctx,
+		      const struct ipv4_frag_id *frag_id,
 		      struct ipv4_frag_l4ports *ports)
 {
 	struct ipv4_frag_l4ports *tmp;
 
 	tmp = map_lookup_elem(&IPV4_FRAG_DATAGRAMS_MAP, frag_id);
-	if (!tmp)
+	if (!tmp) {
+		update_metrics(ctx_full_len(ctx), METRIC_INGRESS,
+			       -DROP_FRAG_NOT_FOUND);
 		return DROP_FRAG_NOT_FOUND;
```

Comment:
When we return an error, we will usually end up dropping the packet and calling `send_drop_notify_error`, which itself calls `send_drop_notify`:
https://github.com/cilium/cilium/blob/1843b4adeddad7ee0fd83bc3c71ba6e4d57db48e/bpf/lib/drop.h#L72-L83
So in this case, we will register two events for `DROP_FRAG_NOT_FOUND`, for a single packet.

I tried following the return path and it looks like in some cases, we simply ignore `DROP_FRAG_NOT_FOUND` returned here (meaning we override it with some other error code). I think we should fix that to ensure we always report the event via `send_drop_notify`.

Reply:
This should now be addressed by the second commit